### PR TITLE
Add Footer component to vue-vocabulary

### LINKED ISSUE #689: [Vue] Footer Component
We need to implement the Footer component from Vocabulary into its Vue counterpart. The main benefits this component will provide are:

- Reduce code boilerplate 
- Prevent users from modifying 'fixed' unchanging footer content (CC's logo, address, etc.)

I would like the initial implementation of the component to be _very limited_ in functionality. We can open up customization with additional sub-components and slots in the future, as more diverse use cases become necessary.

## Screenshot

Screen Shot 2020-09-25 at 11 51 10 AM

## Current HTML version

```html
<footer class="main-footer">
  <div class="container">
    <div class="columns">
      <div class="column is-one-quarter">
        <a href="https://creativecommons.org" aria-label="home" class="main-logo margin-bottom-bigger">
          <span class="has-text-white"><svg
  xmlns="http://www.w3.org/2000/svg"
  preserveAspectRatio="xMidYMid meet"
  viewBox="0 0 304 73">
  <use href="logos/cc/logomark.svg#logomark"></use>
</svg></span>
        </a>
        <div>
          <address class="margin-bottom-normal">Creative Commons<br/>PO Box 1866, Mountain View CA 94042</address>
          <a href="mailto:info@creativecommons.org" class="mail">info@creativecommons.org</a><br/>
          <a href="tel://+1-415-429-6753" class="phone">+1-415-429-6753</a>
        </div>
        <div class="margin-vertical-large">
          <a href="https://www.instagram.com/creativecommons" aria-label="instagram" class="social has-text-white" target="_blank" rel="noopener">
            <i class="icon margin-right-small is-size-4">instagram</i>
          </a>
          <a href="https://www.twitter.com/creativecommons" aria-label="twitter" class="social has-text-white" target="_blank" rel="noopener">
            <i class="icon margin-right-small is-size-4">twitter</i>
          </a>
          <a href="https://www.facebook.com/creativecommons" aria-label="facebook" class="social has-text-white" target="_blank" rel="noopener">
            <i class="icon margin-right-small is-size-4">facebook</i>
          </a>
          <a href="https://www.linkedin.com/company/creative-commons/" aria-label="linkedin" class="social has-text-white" target="_blank" rel="noopener">
            <i class="icon margin-right-small is-size-4">linkedin</i>
          </a>
        </div>
      </div>
      <div class="column is-three-quarters">
        <div class="columns">
          <div class="column is-full">
            <nav aria-label="footerlinks" class="footer-navigation">
              <ul class="menu">
                <li><a href="#" class="menu-item">Item 1</a></li>
                <li><a href="#" class="menu-item">Item 2</a></li>
                <li><a href="#" class="menu-item">Item 3</a></li>
                <li><a href="#" class="menu-item">Item 4</a></li>
                <li><a href="#" class="menu-item">Item 5</a></li>
                <li><a href="#" class="menu-item">Item 6</a></li>
                <li><a href="#" class="menu-item">Item 7</a></li>
              </ul>
            </nav>
          </div>
        </div>
        <div class="columns">
          <div class="column is-two-thirds">
            <div class="subscription">
              <h5 class="b-header">Subscribe to our newsletter</h5>
              <form class="newsletter">
                <label for="email" class="is-sr-only">Email to subscribe</label>
                <input type="text" id="email" class="input" placeholder="Your email">
                <input type="submit" value="subscribe" class="button small">
              </form>
            </div>
            <div class="attribution margin-top-bigger">
              <p class="caption">
                Except where otherwise
                <a href="https://creativecommons.org/policies#license" target="_blank" rel="noopener">noted</a>,
                content on this site is licensed under a
                <a href="https://creativecommons.org/licenses/by/4.0/" target="_blank" rel="noopener">Creative Commons Attribution 4.0 International license</a>.
                <a href="https://creativecommons.org/website-icons" target="_blank" rel="noopener">Icons</a>
                by
                <a href="https://fontawesome.com/" target="_blank" rel="noopener">Font Awesome</a>.
              </p>
              <div class="margin-top-smaller">
                <i class="icon margin-right-small is-size-4">cclogo</i>
                <i class="icon margin-right-small is-size-4">ccby</i>
              </div>
            </div>
          </div>
          <div class="column is-one-third">
            <aside class="donate-section">
              <h5>Our work relies on you!</h5>
              <p>Help us keep the internet free and open.</p>
              <a class="button small donate" href="http://creativecommons.org/donate">
                <i class="icon cc-letterheart-filled margin-right-small is-size-5 padding-top-smaller"></i>
                Donate now
              </a>
            </aside>
          </div>
        </div>
      </div>
    </div>
  </div>
</footer>
```

## Ideal Vue API

The _only_ editable elements of the footer are currently:

- The blue menu items
- Custom main content below the menu items (currently only a newsletter form is used)

The `<FooterMenuItem>` component will use Vue's built in `is` prop (often seen as an `as` prop in react) that changes the actual component rendered, allowing for full control of the html element rendered. We need to update `vocabulary` to make sure none of the styles are dependent on specific html tags, and only on our custom classes.

```vue
  <Footer>
    <!-- The default slot contains the main content of the footer, in this case the newsletter form. -->
    <div class="subscription">
      <h5 class="b-header">Subscribe to our newsletter</h5>
      <form class="newsletter">
        <label for="email" class="is-sr-only">Email to subscribe</label>
        <input type="text" id="email" class="input" placeholder="Your email" />
        <input type="submit" value="subscribe" class="button small" />
      </form>
    </div>

    <!-- Instead of named slots we could wrap named slot implementations in custom components, like <FooterMenuItems>, if preferred -->
    <template #menu-items>
      <FooterMenuItem is="a" href="https://wherever.com">Item 1</FooterMenuItem>
      <FooterMenuItem is="router-link" to="/item-two">Item 1</FooterMenuItem>
    </template>
  </Footer>
```

### PR DESCRIPTION
**Pull request by akmadian**
## Fixes

Fixes #689 by @zackkrida 

## Description

Adds the Footer component to vue-vocabulary and the vue-vocabulary storybook. This implementation allows for customization of the footer links by passing an array of objects with the `links` prop. The schema is as follows:

```
{
    href: String,
    content: String,
    ref: String,
    target: String
}
```
It also includes a slot for the main footer content, this has the newsletter signup form by default.

## Tests

I didn't add any tests since it looks like none have been written for vue-vocabulary yet.

## Screenshots

image

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by dhruvkb — CHANGES_REQUESTED** [review]
Before going into a deeper review, can we import the logos from Vocabulary instead of duplicating the assets in Vue Vocabulary as well?

**Comment by akmadian** [comment]
@dhruvkb done! I don't know why I didn't think of that myself, good catch : )

**Review by dhruvkb — COMMENTED** [review]
This does look like it's leaning towards style B as per the discussion in #642, although I'm not sure if we had decisively chosen style A (since the issue is still open).

File: packages/vue-vocabulary/.storybook/order.js [hunk 1]
```diff
@@ -6,6 +6,7 @@ const order = {
   ],
   'Elements': [],
   'Layouts': [],
+  'Patterns': [],
```
Comment (zackkrida) [inline packages/vue-vocabulary/.storybook/order.js:9]: Great, thanks for doing this

**Review by zackkrida — CHANGES_REQUESTED** [review]
Hi @akmadian this is a great start. As @dhruvkb implied, we do want to go with a 'child component' approach over an array prop for the menu links. You can see some discussion on this decision [here](https://github.com/creativecommons/vocabulary/issues/642) and an example implementation in the [original ticket](https://github.com/creativecommons/vocabulary/issues/689).

Some reasons for doing so:

- A child component approach _can_ be compatible with an array data structure too, so it's more flexible. Example:
```vue
<template>
	<Parent>
		<Child v-for="child in myRandomArray" :href="child.href">
	</Parent>
</template>
```
- It's easier to extend in the future, especially with optional config per-item. Also allows for links that use scoped slots. Example:

```vue
<template>
	<Parent>
		<Child :is="MyCustomLink" href="/cool" :v-slot="{ active }" />
			<CheckmarkIcon v-if="active" /> Cool Page
		</Child>
		<Child is="a" href="https://othersite.com/great" onClick="doExtraAnalyticsThing" />
		<Child is="router-link" to="/my-page" />
	</Parent>
</template>
```

**Comment by akmadian** [comment]
Ahhh gotcha, I didn't catch that thread, will get started on that soon : )

**Comment by obulat** [comment]
I just wanted to add a point about responsiveness: currently, the footer breaks on tablet screens because the rightmost column doesn't fit, and overflows its container:
FooterProblem

I hadn't found any designs for footer on smaller screens, @panchovm, do we have them?.

**Comment by nimishbongale** [comment]
@obulat

I'm not sure about the designs, but here's how the footer looks on smaller screens in the vocabulary `docs` site I'm building:

context

Maybe the same behaviour is expected off of the Vue-Vocabulary Footer component.

At `769x412`, the `opensource.creativecommons.org` footer breaks as well I guess:

context5

Here's the expected behaviour imo:

context4

**Comment by fcoveram** [comment]
I am currently working on the footer's responsiveness and today [I added the mobile version](https://www.figma.com/file/l4Mt3dn3Ndtrvrb4aLcwXI/Design-Library?node-id=639%3A24&viewport=-412%2C204%2C0.27819550037384033). I expect to add the other breakpoints in the following days. Thanks for bringing this @obulat @nimishbongale

**Comment by fcoveram** [comment]
I just created a ticket (#725) to add the responsive versions of footer. The [Design Library page](https://www.figma.com/file/l4Mt3dn3Ndtrvrb4aLcwXI/Design-Library?node-id=639%3A24&viewport=-73%2C237%2C0.23559321463108063) is updated.

**Comment by akmadian** [comment]
I got side tracked by school, I'll work on implementing the child component approach today

**Comment by fcoveram** [comment]
Hi @akmadian 😊 nice to see you again
